BaseClasses.py

```diff
@@ -604,38 +612,157 @@ def all_done() -> bool:
 PathValue = Tuple[str, Optional["PathValue"]]
 
 
-class CollectionState():
-    prog_items: Dict[int, Counter[str]]
+class PlayerState:
+    prog_items: Counter[str]
+    _multiworld: MultiWorld
+    _parent: CollectionState
+    reachable_regions: Set[Region]
+    blocked_connections: Set[Entrance]
+    locations_checked: Set[Location]
+    stale: bool = True
+    player: int
+
+    def __init__(self, player: int, parent: CollectionState):
+        self.prog_items = Counter()
+        self._multiworld = parent.multiworld
+        self._parent = parent
+        self.reachable_regions = set()
+        self.blocked_connections = set()
+        self.locations_checked = set()
+        self.player = player
+
+    def copy(self, state: CollectionState) -> PlayerState:
+        ret = PlayerState(self.player, state)
+        ret.prog_items = self.prog_items.copy()
+        ret.reachable_regions = self.reachable_regions.copy()
+        ret.blocked_connections = self.blocked_connections.copy()
+        ret.locations_checked = self.locations_checked.copy()
+        return ret
+
+    def sweep_for_events(self, key_only: bool = False, locations: Optional[Iterable[Location]] = None) -> None:
+        if locations is None:
+            locations = self._multiworld.get_filled_locations(self.player)
+        self._parent.sweep_for_events(key_only, locations)
+
+    # these signatures need to match the parent so that player state can be used in place for access rule checks
+    def can_reach(
+            self,
+            spot: Union[Location, Entrance, Region, str],
+            resolve_hint: Optional[str] = None,
+            player: Optional[int] = None) -> bool:
```

Comment:
I don't really like keeping the player param here, should just be self.player. The rationale about usage in place doesn't hold because all the has/count methods don't preserve the player param either. It should be better to just get any refactoring effort out of the way now imo

Reply:
When I originally put that comment there I was having worlds create a PlayerState directly and pass that around. It ended up being a lot messier and didn't really improve speed or functionality in any meaningful way, so I scrapped all of that; but I forgot to remove the comment and this spot too it appears.

Reply:
Makes sense